api/handlers.go

```diff
@@ -42,16 +54,55 @@ func CreateSessionHandlerFunc(sessions Sessions) http.HandlerFunc {
 		}
 
 		log.Event(ctx, "session created", log.Data{"session": sess}, log.INFO)
+		s := &session.Session{
+			ID:    sess.ID,
+			Email: sess.Email,
+			Start: sess.Start,
+		}
 
-		sessJson, err := sess.MarshalJSON()
+		if err := cache.Set(s); err != nil {
+			log.Event(ctx, "unable to add session to cache", log.Error(err), log.ERROR)
+			http.Error(w, "unable to add session to cache", http.StatusInternalServerError)
+		}
+		log.Event(ctx, "session added to cache", log.INFO)
+
+		sessJSON, err := sess.MarshalJSON()
 		if err != nil {
-			log.Event(ctx, "failed to marshal session", log.Error(err), log.ERROR)
-			http.Error(w, "Failed to marshal session", http.StatusInternalServerError)
+			failedToMarshalError(ctx, err, w)
 			return
 		}
 
 		w.Header().Set("Content-Type", "application/json")
 		w.WriteHeader(http.StatusCreated)
-		w.Write(sessJson)
+		w.Write(sessJSON)
+	}
+}
+
+func GetByIDSessionHandlerFunc(cache Cache, getVarsFunc GetVarsFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		ID := getVarsFunc(r)["ID"]
+
+		result, err := cache.GetByID(ID)
+		if err != nil {
+			log.Event(ctx,"unable to get session by id", log.Error(err), log.ERROR)
+			http.Error(w, "Unable to get session by id", http.StatusNotFound)
+			return
+		}
+
+		resultJSON, err := result.MarshalJSON()
+		if err != nil {
+			failedToMarshalError(ctx, err, w)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write(resultJSON)
 	}
 }
+
+func failedToMarshalError(ctx context.Context, err error, w http.ResponseWriter) {
+	log.Event(ctx, "failed to marshal session", log.Error(err), log.ERROR)
+	http.Error(w, "Failed to marshal session", http.StatusInternalServerError)
+}
```

Comment:
I think you could take this a step further to make it more resuable

```
func writeErrorResponse(ctx context.Context, w http.ResponseWriter, msg string, err error, status int) {
	log.Event(ctx, msg, log.Error(err), log.ERROR)
	http.Error(w, msg, status)
}
```
You can this use this for any case where you need to write an unsuccessful response.